string base_frame
----
bool success
string error_message
